src/GameLogic/PlayerActions/Items/SimpleItemCraftingHandler.cs

```diff
@@ -208,27 +243,47 @@ private void AddRandomLuckOption(Item resultItem, Player player)
         }
     }
 
-    private void AddRandomExcellentOptions(Item resultItem, Player player)
+    /// <summary>
+    /// Randomly adds skill to the <paramref name="resultItem"/>.
+    /// </summary>
+    /// <param name="resultItem">The result item.</param>
+    /// <param name="successRate">The crafting combination success rate.</param>
+    protected virtual void AddRandomSkill(Item resultItem, byte successRate)
     {
-        if (this._settings.ResultItemExcellentOptionChance > 0
-            && resultItem.Definition!.PossibleItemOptions.FirstOrDefault(o =>
-                    o.PossibleOptions.Any(p => p.OptionType == ItemOptionTypes.Excellent || p.OptionType == ItemOptionTypes.Wing))
-                is { } optionDefinition)
+        if (this._settings.ResultItemSkillChance > 0 && Rand.NextRandomBool(this._settings.ResultItemSkillChance)
+            && !resultItem.HasSkill
+            && resultItem.Definition!.Skill is { })
         {
-            for (int j = 0;
-                 j < optionDefinition.MaximumOptionsPerItem && Rand.NextRandomBool(this._settings.ResultItemExcellentOptionChance);
-                 j++)
-            {
-                var link = player.PersistenceContext.CreateNew<ItemOptionLink>();
-                link.ItemOption = optionDefinition.PossibleOptions
-                    .Except(resultItem.ItemOptions.Select(io => io.ItemOption)).SelectRandom();
-                resultItem.ItemOptions.Add(link);
-                if (resultItem.Definition.Skill != null)
-                {
-                    // Excellent items always have skill.
-                    resultItem.HasSkill = true;
-                }
-            }
+            resultItem.HasSkill = true;
+        }
+    }
+
+    private async ValueTask<List<Item>> CreateResultItemsAsync(Player player, IList<CraftingRequiredItemLink> referencedItems, ItemCraftingResultItem craftingResultItem, byte successRate)
+    {
+        int resultItemCount = this._settings.MultipleAllowed
+            ? referencedItems.FirstOrDefault(r => r.ItemRequirement.Reference > 0)?.Items.Count() ?? 1
+            : 1;
+
+        var resultList = new List<Item>(resultItemCount);
+        for (int i = 0; i < resultItemCount; i++)
+        {
+            // Create new Item
+            var resultItem = player.PersistenceContext.CreateNew<Item>();
+            resultItem.Definition = craftingResultItem.ItemDefinition ?? throw Error.NotInitializedProperty(craftingResultItem, nameof(craftingResultItem.ItemDefinition));
+            resultItem.Level = resultItem.Definition.Group == ItemConstants.Fruits.Group && resultItem.Definition.Number == ItemConstants.Fruits.Number
+                ? (byte)new List<int>() { 0, 1, 2, 3, 4 }.SelectWeightedRandom([30, 25, 20, 20, 5])
```

Comment:
Also interesting. Looking back, this actually makes a lot of sense! Vit and Energy fruits are the most probable, followed by Strength and Agility; Command fruit quite unlikely to come out. At this point in time (S2) almost everyone needs to pump vit, quite some, energy, not so many, str and agi. (At least this is my interpretation of it :smile:).
https://github.com/kyleruss/emu-server/blob/202856a74c905c203b9b2795fd161f564ca8b257/GameServer/Source/ChaosBox.cpp#L1670